cognitedata/reveal
# fix(react-components): don't remove models automatically on switching viewer

#### Type of change

![Bug](https://img.shields.io/badge/Type-Bug-red) 

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/BND3D-4283

## Description :pencil:


This fixes a bug where models would be removed regardless of the new scene content when switching viewer, effectively bypassing the RevealKeepAlive-concept. This is fixed by making sure not to leave the model add data undefined while getting model types.

## How has this been tested? :mag:




## Test instructions :information_source:



In Search, click a 3D asset, then Open to go to its instance page. On the 3D widget, click expand. Note that the model is not re-loaded when entering the expanded widget.


## Checklist :ballot_box_with_check:
